src/main/kotlin/net/atos/zac/healthcheck/HealthCheckService.kt

```diff
@@ -0,0 +1,207 @@
+                ConfiguratieService.STATUSTYPE_OMSCHRIJVING_IN_BEHANDELING ->
+                    zaaktypeInrichtingscheck.isStatustypeInBehandelingAanwezig = true
+                ConfiguratieService.STATUSTYPE_OMSCHRIJVING_HEROPEND ->
+                    zaaktypeInrichtingscheck.isStatustypeHeropendAanwezig = true
+                ConfiguratieService.STATUSTYPE_OMSCHRIJVING_AANVULLENDE_INFORMATIE ->
+                    zaaktypeInrichtingscheck.isStatustypeAanvullendeInformatieVereist = true
+                ConfiguratieService.STATUSTYPE_OMSCHRIJVING_AFGEROND -> {
+                    afgerondVolgnummer = statustype.volgnummer
+                    zaaktypeInrichtingscheck.isStatustypeAfgerondAanwezig = true
+                }
+            }
+        }
+        if (afgerondVolgnummer == hoogsteVolgnummer) {
+            zaaktypeInrichtingscheck.isStatustypeAfgerondLaatsteVolgnummer = true
+        }
+    }
+
+    private fun controleerZaaktypeResultaattypeInrichting(zaaktypeInrichtingscheck: ZaaktypeInrichtingscheck) {
+        val resultaattypes = ztcClientService.readResultaattypen(zaaktypeInrichtingscheck.zaaktype.url)
+        if (resultaattypes.isNotEmpty()) {
+            zaaktypeInrichtingscheck.isResultaattypeAanwezig = true
+            resultaattypes.forEach(
+                Consumer { resultaattype: ResultaatType ->
+                    val afleidingswijzeName = resultaattype.brondatumArchiefprocedure.afleidingswijze.name.lowercase(
+                        Locale.getDefault()
+                    )
+                    // compare enum values and not the enums themselves because we have multiple functionally
+                    // identical enums in our Java client code generated by the OpenAPI Generator
+                    if (Afleidingswijze.VERVALDATUM_BESLUIT.toValue() == afleidingswijzeName ||
+                        Afleidingswijze.INGANGSDATUM_BESLUIT.toValue() == afleidingswijzeName
+                    ) {
+                        zaaktypeInrichtingscheck.addResultaattypesMetVerplichtBesluit(resultaattype.omschrijving)
+                    }
+                }
+            )
+        }
+    }
+
+    private fun controleerZaaktypeBesluittypeInrichting(zaaktypeInrichtingscheck: ZaaktypeInrichtingscheck) {
+        val besluittypes = ztcClientService.readBesluittypen(
+            zaaktypeInrichtingscheck.zaaktype.url
+        ).filter { besluittype: BesluitType -> LocalDateUtil.dateNowIsBetween(besluittype) }
+        if (besluittypes.isNotEmpty()) {
+            zaaktypeInrichtingscheck.isBesluittypeAanwezig = true
+        }
+    }
+
+    private fun controleerZaaktypeRoltypeInrichting(zaaktypeInrichtingscheck: ZaaktypeInrichtingscheck) {
+        val roltypes = ztcClientService.listRoltypen(zaaktypeInrichtingscheck.zaaktype.url)
+        if (roltypes.isNotEmpty()) {
+            roltypes.forEach(
+                Consumer { roltype: RolType ->
```

Comment:
We should get rid of those Java `Consumer` things. Needs a bit of refactoring. :-)